### DISCUSSION TITLE
Inference json output pytorch vs. tensorflow

### DISCUSSION THREAD

**Discussion by kurtgenc** [ultralytics/yolov5#6786] [Q&A]
I'm interested in getting json output for inference using a yolov5 custom model.  When using Tensorflow serving results were formatted as in the example at the bottom.  I'm now using the following code for json output in pytorch. My objective is to format json similar to the tensorflow serving json.

What I need to know is if there's a different format option for json, at least the bounding-box coordinates similar to tf version -- I can try to add the other remaining data elements. The same format is needed so I can use the same json results from both yolov5 and tf.

```
import torch
import json

model = torch.hub.load('ultralytics/yolov5', 'custom', 'runs/train/exp6/weights/best.pt') 

imgs = ['../test/images/IMG_1.jpg']  

results = model(imgs)

results.print()

with open("predict.json", "w") as f:
    f.write(json.dumps(results.pandas().xyxy[0].to_json(orient="records")))

```
Yolov5, pytorch result:
`[{\"xmin\":1558.3446044922,\"ymin\":491.2127990723,\"xmax\":1740.5581054688,\"ymax\":536.5893554688,\"confidence\":0.9156957865,\"class\":0,\"name\":\"g1-0\"},{\"xmin\":756.6607055664,\"ymin\":339.8599853516,\"xmax\":931.6755981445,\"ymax\":385.0133056641,\"confidence\":0.9149582386,\"class\":0,\"name\":\"g1-0\"},{\"xmin\":913.4196777344,\"ymin\":435.6817626953,\"xmax\":1095.7517089844,\"ymax\":481.2907104492,\"confidence\":0.9131733179,\"class\":0,\"name\":\"g1-0\"},{\"xmin\":1678.05859375,\"ymin\":665.1885986328,\"xmax\":1921.4776611328,\"ymax\":709.5936889648,\"confidence\":0.9113134742,\"class\":0,\"name\":\"g1-0\"},{\"xmin\":1539.7135009766,\"ymin\":534.2822265625,\"xmax\":1722.7553710938,\"ymax\":576.6275024414,\"confidence\":0.9109461904 ...
 `
Tensorflow serving result:
```
{
    "predictions": [
        {
            "raw_detection_scores": [[-1.79809, -2.26494, -1.60674, -2.4167, 9.15356, -0.0320532, -0.720658, -2.01969],[33.5114, -2.02846, -2.44318, -7.56883, -1.0466, -9.18447, -8.25208, -3.42913]],
"detection_classes": [1.0, 4.0, 1.0, 4.0, 1.0, 1.0, 4.0, 1.0, 1.0, 1.0, 4.0, 4.0, 1.0, 4.0, 1.0, 1.0, 1.0, 1.0,1.0],
        "num_detections": 30.0,
	"detection_boxes": [[0.11576, 0.189758, 0.2292, 0.340964], [0.362335, 0.764986, 0.530561, 0.929243],[0.176702, 0.745659, 0.443796, 0.837389]],
	"detection_scores": [0.999991, 0.999989, 0.999895, 0.999885, 0.999856, 0.999837, 0.999773, 0.999624]
	}
    ]
}

```

**Comment by kurtgenc** [comment]
To answer my question partially,  it was simple as:
```
df = results.pandas().xyxy[0].drop(columns=['class', 'name'])
df.to_json('../result.json', orient="values")
```
Since the tensorflow json only lists the coordinates, I removed "class" and "name" from the data frame.
Also specified "values" instead of "records".  I used file option, extra write file code is eliminated.
So just 2 lines of code to return the right detection format in json.

**Reply by glenn-jocher** [reply]
@kurtgenc looks good!
